pkg/planner/core/rule/rule_predicate_simplification.go

```diff
@@ -199,15 +288,27 @@ func applyPredicateSimplification(sctx base.PlanContext, predicates []expression
 			simplifiedPredicate = exprs
 		}
 	}
-	simplifiedPredicate = shortCircuitLogicalConstants(sctx, simplifiedPredicate)
-	simplifiedPredicate = mergeInAndNotEQLists(sctx, simplifiedPredicate)
-	removeRedundantORBranch(sctx, simplifiedPredicate)
-	simplifiedPredicate = pruneEmptyORBranches(sctx, simplifiedPredicate)
+
+	eqClasses := buildEquivalenceClasses(simplifiedPredicate)
+	ctx := simplifyContext{
+		planContext:  sctx,
+		ColEqClasses: eqClasses,
+	}
+
+	simplifiedPredicate = inListToOrListAll(sctx, simplifiedPredicate)
+	simplifiedPredicate = shortCircuitLogicalConstants(ctx, simplifiedPredicate)
+	simplifiedPredicate = orList2InListAll(ctx, simplifiedPredicate)
+	simplifiedPredicate = mergeInAndNotEQLists(ctx, simplifiedPredicate)
+	simplifiedPredicate = inListToOrListAll(sctx, simplifiedPredicate)
+	removeRedundantORBranch(ctx, simplifiedPredicate)
+	simplifiedPredicate = pruneEmptyORBranches(ctx, simplifiedPredicate)
 	simplifiedPredicate = constraint.DeleteTrueExprs(exprCtx, sctx.GetSessionVars().StmtCtx, simplifiedPredicate)
+	simplifiedPredicate = orList2InListAll(ctx, simplifiedPredicate)
```

Comment:
Why do we have to call `inListToOrListAll` twice here? Since our goal is to convert all `OR` to `IN`, can we just call `orList2InListAll` here?